codecarbon/external/hardware.py

```diff
@@ -97,14 +97,25 @@ def measure_power_and_energy(
 
     def _get_gpu_ids(self) -> Iterable[int]:
         """
-        Get the Ids of the GPUs that we would like to monitor
+        Get the Ids of the GPUs that we will monitor
         :return: list of ids
         """
+        gpu_ids = []
         if self.gpu_ids is not None:
+            # Check that the provided GPU ids are valid
+            if not set(gpu_ids).issubset(set(range(self.num_gpus))):
+                logger.warning(
+                    f"Unknown GPU ids {gpu_ids}, only {self.num_gpus} GPUs available."
+                )
+            # Keep only the GPUs that are in the provided list
+            for gpu_id in range(self.num_gpus):
+                if gpu_id in self.gpu_ids:
+                    gpu_ids.append(gpu_id)
+                else:
+                    logger.info(
+                        f"GPU number {gpu_id} will not be monitored, at your request."
+                    )
             gpu_ids = self.gpu_ids
-            assert set(gpu_ids).issubset(
-                set(range(self.num_gpus))
-            ), f"Unknown GPU ids {gpu_ids}"
```

Comment:
We stop raising an AssertionError that will break the user code. Instead, we just log a warning above.